added gss to benchmark

bartp5 opened the pull request:
Hi,
Thanks for the great sudoku solver and benchmark tool. 
I created the gss solver (https://github.com/bartp5/gss). I never intended to build such an optimized solver but rather created a solver for many types of sudokus (i.e. jigsaw, X, various sizes, etc.). Just out of curiosity I wanted to see how my solver compares. My solver implements "levels" of logic and uses backtracking when nothing else works, so I used the "configuration" argument to benchmark the various levels of logic.
Cheers,
Bart

bartp5 committed:
added gss to benchmark

t-dillon commented on other/gss/gss.c:
```diff
@@ -0,0 +1,107 @@
+		return 0;
+		
+	sol=malloc(sizeof(MASKINT *));
+	state=Solve(&S, &maxlevel, &sol, &ns, 2, limitlevel, STRAT);
```
You want to pass 'limit' here instead of 2, otherwise this solver won't really support the returns_full_count_ feature and build/run_tests will fail.

t-dillon commented on other/gss/gss.c:
```diff
@@ -0,0 +1,107 @@
+	return 1;
+}
+
+size_t OtherSolverGss(const char *input, size_t limit /* unused */,
```
You can remove both the /*unused*/ here, since you'll use both below.

t-dillon commented on other/gss/gss.c:
```diff
@@ -0,0 +1,107 @@
+    Generic Sudoku Solver (gss) link to tdoku benchmark
+    Copyright (C) 2019 B. Pieters
+
+    This program is free software: you can redistribute it and/or modify
```
Can you change the license on this file to a permissive one? Everything I'm actually including in the project is BSD or similar.

t-dillon reviewed (changes requested):
Hi Bart, thanks for adding in gss. Looks like a cool project!

See a few comments below.

Also, please add a bullet for gss in other/README.md

Thanks!

bartp5 committed:
- changed license for gss.c to BSD
- in gss.c calling routine, use limit for what it is intended for
- added gss to the list in other/README

bartp5 commented:
All done.
Thanks

bartp5 committed:
- free memory in case sudoku is invalid

bartp5 committed:
added gss to the solvers listed in the README

t-dillon commented:
Thanks Bart!
